prisma/schema.prisma

```diff
@@ -11,14 +11,111 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
-model Log {
-  id      String @id @default(uuid())
-  level   Level
-  message String
+model Ship {
+  id      String  @id @default(uuid())
+  name    String  @unique
+  // maneuvers Maneuver[] // I don't know how we want to handle custom types here, because what I want is a {ManeueverType, Difficulty} tuple, e.g. {"5K", "Red"}
+  // If we ever hit the case of a chassis having different hull/shields (1.0 YT-1300), we'd have to move these to the pilot
+  hull    Int
+  shields Int
+  // arcs Arc[] // Same as maneuevers - I want tuples and I don't know the right way to do it with typing, e.g. [{"front", "2'}, {"bullseye", "3'}]
+  pilots  Pilot[]
 }
 
-enum Level {
-  Info
-  Warn
-  Error
+model ShipAbility {
+  id    String  @id @default(uuid())
+  name  String
+  text  String
+  Pilot Pilot[]
+}
+
+model Pilot {
+  id             String        @id @default(uuid())
+  faction        String
+  shipId         String
+  ship           Ship          @relation(fields: [shipId], references: [id])
+  name           String
+  subtitle       String
+  initiative     Int
+  force          Int // XWS has this as a bag with the force and forceRecover packed... I don't have a gut sense of what I like better but split out for stronger typing for now
+  forceRecover   Int
+  charges        Int // Same as force, packed in XWS
+  chargesRecover Int
+  limit          Int // Fighting my inner pedant here - more technically correct feels like a "limited" boolean and then a nullable "limit" int, but a limit of 0 accomplishes the same
+  pilotAbility   String
+  shipAbilityId  String
+  shipAbility    ShipAbility   @relation(fields: [shipAbilityId], references: [id])
+  PilotPoints    PilotPoints[]
+
+  @@unique([faction, name, subtitle])
+}
+
+model Upgrade {
+  id    String        @id @default(uuid())
+  name  String
+  types UpgradeType[]
+  text  String
+
+  @@unique([name, types])
+  PilotPoints   PilotPoints?    @relation(fields: [pilotPointsId], references: [id])
+  pilotPointsId String?
+  UpgradePoints UpgradePoints[]
+}
+
+model PointsEnvironment {
+  id             String          @id @default(uuid())
+  releaseDate    DateTime?
+  effectiveDate  DateTime?
+  source         String
```

Comment:
I wonder if source here would be better as some map or enum to different sources. Or, if it's only going to support AMG initially anyways, it may be better to omit and save for a version 2?

Reply:
Good point, enum is probably better. Omit and save seems good to me, kick the can since these are going to be so few that they'll be easy to update if we ever decide it's necessary. Will update.